### PR TITLE
feat(BUX-5, BUX-176): Merkle Proofs and Compound Merkle Path

### PR DESCRIPTION
**Pull request by vitalibalashka**
- Retrieving and storing of Merkle Proof from mAPI within sync-task
- Calculation and storing of Compound Merkle Path

### CODE REVIEW

File: model_draft_transactions_test.go [hunk 1]
```diff
@@ -235,7 +233,7 @@ func TestDraftTransaction_createTransaction(t *testing.T) {
 	})
 
 	t.Run("transaction with utxos", func(t *testing.T) {
-		ctx, client, deferMe := CreateTestSQLiteClient(t, false, false, WithCustomTaskManager(&taskManagerMockBase{}))
+		ctx, client, deferMe := CreateTestSQLiteClient(t, false, true, WithCustomTaskManager(&taskManagerMockBase{}))
```
Comment (dorzepowski) [inline model_draft_transactions_test.go:236]: Why this change?

File: model_merkle_proof.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	marshal, err := json.Marshal(m)
+	if err != nil {
+		return nil, err
+	}
+
+	return string(marshal), nil
+}
+
+// ToCompoundMerklePath transform Merkle Proof to Compound Merkle Path
+func (m MerkleProof) ToCompoundMerklePath() CompoundMerklePath {
```
Comment (dorzepowski) [inline model_merkle_proof.go:50]: Please move it up, as this is more important then Scan and Value methods.

File: model_merkle_proof.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+	if height == 0 {
+		return nil
+	}
+	cmp := make(CompoundMerklePath, height)
+	leafMap := make(map[string]uint64, 2)
+	offset := m.Index
+	leafMap[m.TxOrID] = offset
+	leafMap[m.Nodes[0]] = offsetPair(offset)
+	cmp[0] = leafMap
+	for i := 1; i <= height-1; i++ {
```
Comment (dorzepowski) [inline model_merkle_proof.go:61]: ### Minor
```suggestion
	for i := 1; i < height; i++ {
```

File: model_merkle_proof.go [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+		leaf := make(map[string]uint64, 1)
```
Comment (dorzepowski) [inline model_merkle_proof.go:62]: ## Minor

It isn't a leaf every time,
I think that it should be path or node (path seems to be better because it doesn't collide with BSV Node term)

File: model_draft_transactions_test.go (hunk 1 above)
Comment (vitalibalashka) [reply model_draft_transactions_test.go:236]: From now processing of utoxos checks transactions where these utxos are came from.
So for test cases I need to create a test transaction.
And with "shared" flag as false there will be initiated different SQLiteTestConfig with different tablePrefixes. And it's a crutual for these tests as we save destination with one config first and then we try to save a tx with different config(which has different table prefix) and it can't find destination as destination table with such prefix does not exist

File: model_merkle_proof.go (hunk 3 above)
Comment (vitalibalashka) [reply model_merkle_proof.go]: My bad. Fixed

File: model_merkle_proof.go (hunk 4 above)
Comment (vitalibalashka) [reply model_merkle_proof.go]: Renamed to pathMap, thanks

File: model_merkle_proof.go (hunk 2 above)
Comment (vitalibalashka) [reply model_merkle_proof.go]: Done

Comment (dorzepowski) [reply model_merkle_proof.go]: To high, it should be after:
```
// MerkleProof represents Merkle Proof type
type MerkleProof bc.MerkleProof
```

but before `Scan` and `Value`

File: model_merkle_proof.go (hunk 4 above)
Comment (dorzepowski) [reply model_merkle_proof.go]: still this one `leaf := make(map[string]uint64, 1)` is named leaf

File: model_merkle_proof.go (hunk 2 above)
Comment (vitalibalashka) [reply model_merkle_proof.go]: OMG. Of course. It's all the aura of the coming weekend

Comment (vitalibalashka) [reply model_merkle_proof.go]: Fixed

File: model_merkle_proof.go (hunk 4 above)
Comment (vitalibalashka) [reply model_merkle_proof.go]: No more leafs. Only paths

**Review by dorzepowski — APPROVED** [review]

**Comment by vitalibalashka** [comment]
create a PR to BuxOrg https://github.com/BuxOrg/bux/pull/381
